src/SequenceManager.cc

```diff
@@ -358,4 +397,56 @@ void SequenceManager::store_sequence(const Sequence& seq) const
         store_step(seq_path / extract_filename_step(++idx, max_digits, step), step);
 }
 
+std::string SequenceManager::stage_files_in_directory(std::filesystem::path dir_name, const std::string& filetype)
+{
+
+    // detect what has changed in the sequence
+    std::string git_msg{""};
+    auto stats = gl_.status();
+    for(const auto& elm: stats)
+    {
+        
+        // filter for changes in sequence
+        if (gul14::starts_with(elm.path_name, (std::string) dir_name))
+        {
+            if (elm.changes == "new file" and (filetype == elm.changes or filetype == ""))
```

Comment:
I do not fully understand what you mean, I think I need an explanation in person

Reply:
I mean something like this:

Class `GitRepository`:

```c++

namespace git {

enum class FileTypeChange
{
    new_file,
    deleted,
    ...
};

struct FileStatus
{
    std::string path_name; NEW_NAME".
    std::string handling;
    FileTypeChange changes;
};

...
} // namespace git
```

Class `SequenceManager`:

```c++
std::string SequenceManager::stage_files_in_directory(std::filesystem::path dir_name, FileTypeChange filetype)
{
    ...
            if (elm.changes == FileTypeChange::new_file and filetype == elm.change)
```

Reply:
I agree, it would be nice if `libgit4cpp` could take care of parsing the strings and just hand some enums to its users. Maybe we can make an issue out of that and address it later?

Reply:
Removed the parsing completely.